src/main/scala/no/ndla/articleapi/repository/ArticleRepository.scala

```diff
@@ -169,37 +170,23 @@ trait ArticleRepository {
       }
     }
 
-    def applyToAll(func: (List[Article]) => Unit)(implicit session: DBSession = AutoSession): Unit = {
-      val (minId, maxId) = minMaxId
-      val groupRanges = Seq.range(minId, maxId).grouped(ArticleApiProperties.IndexBulkSize).map(group => (group.head, group.last + 1))
-      val ar = Article.syntax("ar")
-
-      groupRanges.foreach(range => {
-        func(
-          sql"select ${ar.result.*} from ${Article.as(ar)} where ${ar.id} between ${range._1} and ${range._2}".map(Article(ar)).toList.apply
-        )
-      })
-    }
-
     def all(implicit session: DBSession = AutoSession): List[Article] = {
       val ar = Article.syntax("ar")
-      sql"select ${ar.result.*} from ${Article.as(ar)}".map(Article(ar)).list.apply()
+      sql"select ${ar.result.*} from ${Article.as(ar)} where document is not NULL".map(Article(ar)).list.apply()
     }
 
-    def allWithExternalSubjectId(externalSubjectId: String): Seq[Article] =
-      articlesWhere(sqls"$externalSubjectId=ANY(ar.external_subject_id)")
+    def allWithExternalSubjectId(externalSubjectId: String): Seq[Article] = articlesWhere(sqls"$externalSubjectId=ANY(ar.external_subject_id)")
```

Comment:
Burde ikke denne ha samme `...where document is not NULL` clause som resten?

Reply:
Jepp, den bør den. Og det har den allerede :) `is not NULL` legges på i `articlesWhere`-funksjonen